test/test_torch.py

```diff
@@ -12665,6 +12665,25 @@ def test_fill_diagonal(self):
         e1.fill_diagonal_(v, wrap=True)
         self.assertEqual(e1, e2)
 
+    def test_binary_op_input_output_overlap(self):
+        sz = 3
+        data = torch.randn(3 * sz)
+        for i in range(len(data) - sz):
+            for j in range(len(data) - sz):
+                for k in range(len(data) - sz):
+                    a = data[i:i + sz]
+                    a_exp = a.clone()
+                    b = data[j:j + sz]
+                    b_exp = b.clone()
+                    c = data[k:k + sz]
+                    c_exp = c.clone()
+                    torch.add(a_exp, b_exp, out=c_exp)
+                    try:
+                        torch.add(a, b, out=c)
+                        self.assertEqual(c_exp, c)
+                    except RuntimeError:
+                        pass
```

Comment:
Would be good to assert that the error message matches what the error message in the code